src/main/java/com/esri/core/geometry/ogc/OGCConcreteGeometryCollection.java

```diff
@@ -867,24 +864,23 @@ public OGCConcreteGeometryCollection flatten() {
 	/**
 	 * Fixes topological overlaps in the GeometryCollecion.
 	 * This is equivalent to union of the geometry collection elements.
-	 * 
+	 *
+	 * TODO "flattened" collection is supposed to contain only mutli-geometries, but this method may return single geometries
```

Comment:
removeOverlapsHelper_ should return List<OGCGeometry> and it should call convertToMulti on each element to fix this.